glitter-bug/Bangazon-Platform-API
# LPbranch1

# Description

Customer ticket, needed to add methods to GET all Get(id), POST, and PUT, as well as include specific query string parameters.

Fixes # (issue)

**GET**
**POST**
**PUT**

User should be able to GET a list of customers, and GET a single customer.
* If the query string parameter of `?_include=products` is provided, then any products that the customer is selling should be included in the response.
* If the query string parameter of `?_include=payments` is provided, then any payment types that the customer has used to pay for an order should be included in the response.
* If the query string parameter of `q` is provided when querying the list of customers, then any customer that has property value that matches the pattern should be returned.

## Type of change

# Testing Instructions
Please test to see if 
`/api/customers/"insert a number here"?_include=payments`
`/api/customers/"insert a number here"?_include=products`
`/api/customers?q="insert a letter here"`
 work as intended if the CustomerId does not have a PaymentType or a ProductType it should display all customers.

- [x] SQL Script runs



# Checklist:

- [x] I have written tests for all of the code
